extensions/wikia/PortableInfobox/services/Parser/Nodes/Node.php

```diff
@@ -55,10 +42,17 @@ public function getExternalParser() {
 	}
 
 	/**
-	 * @param mixed $externalParser
+	 * @param ExternalParser|null $externalParser
+	 *
+	 * @return $this
 	 */
-	public function setExternalParser( ExternalParser $externalParser ) {
-		$this->externalParser = $externalParser;
+	public function setExternalParser( $externalParser ) {
+		// we can pass anything, and ignore it if not ExternalParser instance
```

Comment:
We use builder pattern here, for fluently passing external parser to children nodes. Type hinting was removed to prevent catchable fatal error appearing.